src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -437,8 +443,40 @@ def _filter_privacy_request_queryset(
                 ),
             ).values(column("privacy_request_id"))
         }
-        query = query.filter(PrivacyRequest.id.in_(identities))
-    # Further restrict all PrivacyRequests by query params
+        query = query.filter(PrivacyRequest.id.in_(identity_set))
+
+    if identities:
+        identity_conditions = [
+            (ProvidedIdentity.field_name == field_name)
+            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(value))
+            for field_name, value in identities.items()
+        ]
+
+        identities_query = select([ProvidedIdentity.privacy_request_id]).where(
+            or_(*identity_conditions)
+            & (ProvidedIdentity.privacy_request_id.isnot(None))
+        )
+        query = query.filter(PrivacyRequest.id.in_(identities_query))
+
+    if custom_privacy_request_fields:
+        custom_field_conditions = [
+            (CustomPrivacyRequestField.field_name == field_name)
+            & (
+                CustomPrivacyRequestField.hashed_value
+                == CustomPrivacyRequestField.hash_value(value)
+            )
+            for field_name, value in custom_privacy_request_fields.items()
+        ]
+
+        custom_fields_query = select(
+            [CustomPrivacyRequestField.privacy_request_id]
+        ).where(
+            or_(*custom_field_conditions)
+            & (CustomPrivacyRequestField.privacy_request_id.isnot(None))
+        )
+        query = query.filter(PrivacyRequest.id.in_(custom_fields_query))
```

Comment:
This is the key part of this change 🔑

Reply:
Could we maybe add to the docstring that identities and and custom fields are "or" filters?